### PR TITLE
fix logic of HasNext. Pre-init search by filename. Update CSS for category template.

### PR DESCRIPTION
**Pull request by wipedlife**

### CODE REVIEW

File: src/tracker/db/postgres.go [hunk 1]
```diff
@@ -131,6 +131,16 @@ func (st *Postgres) FindTorrentByInfohash(ih [20]byte) (t *model.Torrent, err er
 	return
 }
 
+func (st *Postgres) FindTorrentsByFile(name string, perpage, offset int) (torrents []model.Torrent, err error){
+	var rows *sql.Rows
+	rows, err = st.conn.Query(fmt.Sprintf("SELECT id, filename, meta_infohash FROM %s WHERE filename LIKE '%$1%' ORDER BY filename DESC LIMIT $2 OFFSET $3", tableFiles), name, perpage, offset)
+	if err == nil{
+		for rows.Next(){
```
Comment (majestrate) [inline src/tracker/db/postgres.go:138]: this doesn't seem to be done yet

Comment (wipedlife) [reply src/tracker/db/postgres.go:138]: Pre-init search by filename

**Comment by wipedlife** [comment]
X nice, that looks better.
X can you see the placeholders on frontpage template for seed count? 
I torrents.html.tmpl?
I which stats?
X frontpage.html.tmpl
I aa
X               <td colspan="5"><b>Recent Uploads</b></td>-->
X               <th>Seeds</th>-->
X               <td> .Complete </td>-->
X that's where we want the seed count to be listed.
I i can did array with that
I for 
I {{ range .SeedsCount }}
X and same for /s/?id=
I shitcoding top master aka livingstone 
X lol
X chocolate star for you!
I search 
I ok
X only when search returns results, obviously, not for the Popular Tags table. 
I err = s.tmpl.ExecuteTemplate(w, "frontpage.html.tmpl", map[string]interface{}{ ...
X it looks like the SeedsCount array needs to be able to be referenced from the .Torrents array? 
X not sure how nested ranges work in go..
I i too
I torrents array its another array
I maybe its struct
X I tried nesting before I think and it didn't work.
I type Torrent struct {
I i think more easy way its download scrape file -> read -> fixmap -> use that array of Scrape struct for template
I but 
I we have to use the array in torrents range
X sure, sounds about right. 
I torrents, err := s.DB.GetFrontPageTorrents()
I torrents []model.Torrent
I jes its struct
I we can add in struct of Torrent var ... []Scrape
I and use I i think more easy way its download scrape file -> read -> fixmap ->... 
I or add seeds/peers/... in db
I and update db seeds/peers/.. in gorouite 
X your call, livingstone.. whatever seems best.
I or every time download scrape file = load zzzot server = lags
X yeah, scrapes should be persisted somewhere fo' sure.
X and maybe scrapes get run on a schedule, every 15 minutes?
I jes

File: src/tracker/db/postgres.go [hunk 2]
```diff
@@ -369,7 +369,7 @@ func (db *Postgres) GetCommentsForTorrent(t *model.Torrent) (comments []model.Co
 
 func (db *Postgres) GetTorrentTags(t *model.Torrent) (tags []model.Tag, err error) {
 	var rows *sql.Rows
-	rows, err = db.conn.Query(fmt.Sprintf("SELECT id, name, tagrank(id) as rank FROM %s WHERE id IN ( SELECT tag_id from %s WHERE tag_infohash = $1) ORDER BY name DESC", tableTags, tableTagMetaInt), t.InfoHash())
+	rows, err = db.conn.Query(fmt.Sprintf("SELECT id, name, tagrank(id) as rank FROM %s WHERE id IN ( SELECT tag_id from %s WHERE tag_infohash = $1) ORDER BY name", tableTags, tableTagMetaInt), t.InfoHash())
```
Comment (majestrate) [inline src/tracker/db/postgres.go:372]: this changes how the query sorts the results in a way that is not correct

**Comment by wipedlife** [comment]
its hard fork now.

**Pull request closed without merging**